### ISSUE TITLE
Empty string in nested object of entity, is retrieved as an empty object.

### ISSUE THREAD

**Issue by alexd6631** [telefonicaid/fiware-orion#1642]
Hello,

When using nested objects in entities, I have encountered the following behavior :

Given the following entity creation
```
POST /v1/updateContext HTTP/1.1
Content-Type: application/json
Accept: application/json

{
    "contextElements": [
        {
            "type": "MyEntity",
            "isPattern": "false",
            "id": "1",
            "attributes": [
                {
                    "name": "name",
                    "type": "string",
                    "value": "test"
                },
                {
                    "name": "empty",
                    "type": "string",
                    "value": ""
                },
                {
                    "name" : "objectWithEmptyString",
                    "type" : "T",
                    "value" : [{
                        "key": ""
                    }]
                }
            ]
        }
    ],
    "updateAction": "APPEND"
} 
```

When I query for this entity

```
POST /v1/queryContext HTTP/1.1
Content-Type: application/json
Accept: application/json
{
    "entities": [
        {
            "type": "MyEntity",
            "isPattern": "true",
            "id": ".*"
        }
    ]
}
```
The result obtained is 
```json
{
  "contextResponses": [
    {
      "contextElement": {
        "type": "MyEntity",
        "isPattern": "false",
        "id": "1",
        "attributes": [
          {
            "name": "empty",
            "type": "string",
            "value": ""
          },
          {
            "name": "name",
            "type": "string",
            "value": "test"
          },
          {
            "name": "objectWithEmptyString",
            "type": "T",
            "value": [
              {
                "key": {}
              }
            ]
          }
        ]
      },
      "statusCode": {
        "code": "200",
        "reasonPhrase": "OK"
      }
    }
  ]
}
```

The expected response would contain an empty string, such as
```
{ ...
"value": [{ "key": "" }]
...}
```

Is this behavior normal and part of the NGSI specification ?
Thanks

**Comment by fgalan** [comment]
No, this is not normal behaviour. Actually, it seems to be a bug that cause that empty strings (i.e. `""`) in internal keys in compound attribute values is interpreted as `{}` instead.

Thanks for the feedback!

**Comment by fgalan** [comment]
After reseraching on this, it seems that it happens in the update stage due to the compound attributes parsing logic wrongly detect the "key" node as an object compound (instead of string).

In the DB has a correct key (hacking at DB level to have a valid document) it has been found that the GET operation on the entity returns the correct document.

The following .test file should be included as part of the PR solving this issue: https://gist.github.com/fgalan/e28436ea6aef9e0f5101

**Comment by fgalan** [comment]
Fixed in PR #1660
